app/stac_api/serializers.py

```diff
@@ -933,7 +933,7 @@ class UploadPartSerializer(serializers.Serializer):
     # pylint: disable=abstract-method
     etag = serializers.CharField(source='ETag', allow_blank=False, required=True)
     part_number = serializers.IntegerField(
-        source='PartNumber', min_value=1, max_value=10000, required=True, allow_null=False
+        source='PartNumber', min_value=1, max_value=100, required=True, allow_null=False
```

Comment:
Does the max file size limit also need to be checked?

Reply:
Well this value should not changed because of the lack of pagination for the urls that depend on this number. And also if we would do it here, then we should do for all validator parameters in models and serializer. I think we should know that model and serializer must remain in sync and we don't need to define variables for this for simplicity.

Reply:
Makes sense, I guess you mean my comment on the gobal setting or env var for this.
What about checking the max allowed file size? Does that need to be validated for each part?

Reply:
The actual upload is not done via our api, but directly with the S3 API through the presigned URL. So we have no control about the file size at all. However if the file size are not in the allowed range; 5MB - 5GB for each parts except the last one which don't have a minimum, S3 will complain with a 400 and a proper message. I already added comment about file size in the spec in the next coming PR.

Reply:
Another stupid question: If I understood correctly, the value for `max_value` here depends on our `PAGE_SIZE` settings, right? Then we could probably simply use `os.environ.get('PAGE_SIZE', 100)` here to be consistent.

Reply:
There is no dependency with the page size. The size of `urls` array depends on the `number_parts` and we don't have any pagination for the array `urls`. That's why I limited the `number_parts` to 100 has it would allow to upload a file up to 500GB (a part cannot be more than 5GB) and the answer of the create endpoint would still have an acceptable size. Originaly I set the `number_parts` to the S3 limit which was 10'000 but this would have cause trouble because we would then loop 10'000 times to create presigned url and the create multipart upload endpoint answer would have had a 10'000 size `urls` array.

Reply:
OK, so i misinterpreted "pagination", sorry ;-)